### PR TITLE
[Enhancement] - FireEye HX

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-448?filter=-1)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py [hunk 1]
```diff
@@ -830,6 +830,87 @@ def get_list_containment_request(self, offset: int, limit: int, state_update_tim
...
+                    'add': host_ids_to_add,
+                    'remove': host_ids_to_remove
+                }
+            ]
+        }
+
+        return body
+
+    @staticmethod
+    def create_dynamic_host_body_request(host_set_name: str, query: str, query_key: str, query_value: str, query_operator: str):
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:896]:
```suggestion
    def create_dynamic_host_request_body(host_set_name: str, query: str, query_key: str, query_value: str, query_operator: str):
```

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py [hunk 2]
```diff
@@ -1962,6 +2043,155 @@ def cancel_containment_command(client: Client, args: Dict[str, Any]) -> CommandR
...
+    host_set_id = args.get('host_set_id')
+
+    if not host_set_id:
+        raise ValueError('Host Set ID is required')
+
+    try:
+        client.delete_host_set_request(host_set_id)
+        message = f'Host set {host_set_id} was deleted successfully'
+    except Exception as e:
+        if '404' in str(e):
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2061]: just to make sure: status code will be 200 even when it's actually a failure?

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py [hunk 3]
```diff
@@ -1962,6 +2043,155 @@ def cancel_containment_command(client: Client, args: Dict[str, Any]) -> CommandR
...
+    hosts_ids = argToList(args.get('hosts_ids'))
+
+    if not hosts_ids:
+        raise ValueError('Please provide at least one Host ID.')
+    elif not host_set_name:
+        raise ValueError('Host Set name is required.')
+    data = None
+    try:
+        response = client.create_static_host_set_request(host_set_name, hosts_ids)
+        data = response['data']
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2080]: let's make sure data exists in the response, something like:
```suggestion
        if not (data := response.get('data')):
            <do what you need to do here>
```

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.yml [hunk 4]
```diff
@@ -1687,21 +2121,265 @@ script:
...
+    - default: false
+      description: The host set id.
+      isArray: false
+      name: host_set_id
+      required: true
+      secret: false
+    deprecated: false
+    description: Deletes a host set.
+    execution: true
+    name: fireeye-hx-delete-host-set
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.yml:2153]: let's add for this command the outputs: `FireEyeHX.HostSets._id` and `FireEyeHX.HostSets.deleted`
where deleted is true, of course.
in other commands, we will have this path too, where `deleted` is `false`. (we did it with Sentinel too if you remember)

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2_test.py [hunk 5]
```diff
@@ -1664,3 +1664,177 @@ def test_list_indicator_categories_command(mocker, requests_mock, file_name: str
     assert command_result.to_context() == expected_context
     assert request.called_once
     assert request.last_request._url_parts.query == 'limit=49&search=foo'
+
+
+def test_delete_host_set_command(mocker):
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2_test.py:1669]: please add given-when-then where needed

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.yml [hunk 6]
```diff
@@ -1687,21 +2121,265 @@ script:
...
+      type: Date
+  - arguments:
+    - default: false
+      description: The host set name.
+      isArray: false
+      name: host_set_name
+      required: true
+      secret: false
+    - default: false
+      description: Free text query. Cannot be used with the other query arguments.
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.yml:2192]: let's add here a link to the documentation explaining how to build the query.

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py [hunk 7]
```diff
@@ -1962,6 +2043,155 @@ def cancel_containment_command(client: Client, args: Dict[str, Any]) -> CommandR
...
+    query_key = args.get('query_key')
+    query_value = args.get('query_value')
+    query_operator = args.get('query_operator')
+
+    if not host_set_name:
+        raise ValueError('Host Set name is required.')
+    elif query and (query_key or query_value or query_operator):
+        raise ValueError('Cannot use free text query with other query operators, Please use one.')
+    elif not (query_key and query_value and query_operator) and not query:
+        raise ValueError('Please provide a free text query, or add all of the query operators toghether.')
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2139]: you need to check here that all of query_key, query_value and query_operator are used.

**Review by dantavori — CHANGES_REQUESTED** [review]
nice job,
some of my comments in the code are relevant to other commands but I haven't written them again.
let's schedule a demo (only with me) for next week.

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md [hunk 8]
```diff
@@ -2196,30 +2196,29 @@ Create a new indicator.
 >| YYYXXXYYY |
 
 
-### fireeye-hx-delete-host-set-policy
+### fireeye-hx-delete-host-set
```
[the remark this answers is not in the document]
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md:2199]: image
It's actually a different command that was missing from the README file

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py (hunk 2 above)
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2061]: image
According to the documentation, those are the only 2 error codes that can return

Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2061]: image
According to the documentation, there are only 2 possible status codes

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py [hunk 9]
```diff
@@ -830,6 +830,87 @@ def get_list_containment_request(self, offset: int, limit: int, state_update_tim
...
+        return body
+
+    @staticmethod
+    def create_dynamic_host_body_request(host_set_name: str, query: str, query_key: str, query_value: str, query_operator: str):
+        body = {
+            'name': host_set_name,
+        }
+
+        if query:
+            body['query'] = query
```
[the remark this answers is not in the document]
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:902]: Will check if the query should be sent as json or string

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py (hunk 3 above)
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2080]: If the request succeeded, there is always a data entry, but I'll put in that check

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:902]: Checked and it needs to be a JSON object. will change

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py (hunk 3 above)
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2.py:2080]: did you mean `if data := response.get('data'):` ?

**Comment by ShirleyDenkberg** [comment]
@dantavori Doc review completed.

File: Tests/conf.json [hunk 10]
```diff
@@ -5451,7 +5451,8 @@
         "Alibaba Action Trail Event Collector": "XSIAM integration",
         "Salesforce Event Collector": "XSIAM integration",
         "Github Event Collector" : "XSIAM integration",
-        "Microsoft Defender for Cloud Apps Event Collector": "XSIAM integration"
+        "Microsoft Defender for Cloud Apps Event Collector": "XSIAM integration",
+        "FireEye HX": "the integration is deprecated as we released FireEyeHX v2"
```
Comment (dantavori) [inline Tests/conf.json:5455]: better to remove it from the conf.json, rather than skip

File: Packs/FireEyeHX/ReleaseNotes/2_2_6.md [hunk 11]
```diff
@@ -1,5 +1,6 @@
+
```
Comment (dantavori) [inline Packs/FireEyeHX/ReleaseNotes/2_2_6.md:1]: make sure this file is not changed at all.

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md [hunk 12]
```diff
@@ -2211,16 +2213,6 @@ Deletes a Host Set policy.
...
-
-#### Context Output
-
-There is no context output for this command.
-#### Command example
-```!fireeye-hx-delete-host-set-policy hostSetId=1005 policyId=YYYXXXYYY```
-#### Human Readable Output
-
->Success
-
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md:2223]: why deleted?

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2_test.py [hunk 13]
```diff
@@ -1664,3 +1664,225 @@ def test_list_indicator_categories_command(mocker, requests_mock, file_name: str
...
+    from FireEyeHXv2 import update_dynamic_host_set_command
+    client = ""
+
+    with pytest.raises(Exception) as e:
+        update_dynamic_host_set_command(client, args)
+    assert str(e.value) == expected_results
+
+
+def test_create_dynamic_host_request_body(mocker):
+
```
Comment (dantavori) [inline Packs/FireEyeHX/Integrations/FireEyeHXv2/FireEyeHXv2_test.py:1855]: add given-when-then please (also below)

**Review by dantavori — APPROVED** [review]

File: Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md (hunk 12 above)
Comment (MLainer1) [reply Packs/FireEyeHX/Integrations/FireEyeHXv2/README.md:2223]: Probably happened during the doc review, will add it back

**Pull request merged**